### PR TITLE
Names the Effort enum of QuantConflictFind class.

### PR DESCRIPTION
**Pull request by timothy-king**
Adding a third value to the enum to indicate not being set. Minor refactoring related to this change. Resolves CID 1172043.

### CODE REVIEW

**Comment by ajreynol** [comment]
This class is the most performance intensive and critical classes in all of quantifiers unfortunately, so I would like to performance test these changes.

File: src/theory/quantifiers/sygus_process_conj.h [hunk 1]
```diff
@@ -353,8 +353,7 @@ class CegConjectureProcess
                          NodeHashFunction>& free_vars);
   /** for each synth-fun, information that is specific to this conjecture */
   std::map<Node, CegConjectureProcessFun> d_sf_info;
-  /** reference to quantifier engine */
-  QuantifiersEngine* d_qe;
+
```
Comment (ajreynol) [inline src/theory/quantifiers/sygus_process_conj.h]: I have downstream changes that are incompatible with this.

**Comment by ajreynol** [comment]
Actually on second thought, I think I'd rather address this myself.  I assume that coverity is complaining about not using an enum?  Perhaps you can give me a pointer?

File: src/theory/quantifiers/sygus_process_conj.h (hunk 1 above)
Comment (timothy-king) [reply src/theory/quantifiers/sygus_process_conj.h]: Right now it is dead code and is triggering warnings in clang. If we need it later, we can add it back.

**Comment by timothy-king** [comment]
> This class is the most performance intensive and critical classes in all of quantifiers unfortunately, so I would like to performance test these changes.

Feel free to try it out. The only change that is likely to have a measure could be getQcfEfforts(), which could be further optimized if needed.

> I assume that coverity is complaining about not using an enum? Perhaps you can give me a pointer?

CID #1172043. The problem is that d_effort was not previously initialized.

> Actually on second thought, I think I'd rather address this myself.

How would you like to address this? I can try to address the concerns in the pull request.

**Comment by ajreynol** [comment]
I'll think about this and get back to you.  Btw, #1366 is not related to this.

File: src/theory/quantifiers/quant_conflict_find.cpp [hunk 2]
```diff
@@ -2061,9 +2065,8 @@ void QuantConflictFind::check( Theory::Effort level, unsigned quant_e ) {
         debugPrint("qcf-debug");
         Trace("qcf-debug") << std::endl;
       }
-      short end_e = getMaxQcfEffort();
       bool isConflict = false;
-      for( short e = effort_conflict; e<=end_e; e++ ){
+      for (Effort e : getQcfEfforts()) {
```
Comment (ajreynol) [inline src/theory/quantifiers/quant_conflict_find.cpp]: I don't want to change getQcfEffort to a vector. That is my main complaint about this PR.
Btw, I don't know your preferred way of iterating on enums, but this is what I did in a recent pull request:

https://github.com/CVC4/CVC4/commit/85df7998e4362e0a9c796146d07d7b9e91045a31#diff-d66c6276213ae410fc43f004a2dd20a9R562

I'd like to do something similar here.

Comment (timothy-king) [reply src/theory/quantifiers/quant_conflict_find.cpp]: Removed the getQcfEffort() vector and switched back to a similar way of iterating through the values.

**Comment by timothy-king** [comment]
PTAL

**Pull request merged**